scripts/test-build-config.ps1

```diff
@@ -89,7 +87,17 @@ if ($ADD_WINDOWS_ENCLAVE_TESTS) {
     $ADD_WINDOWS_ENCLAVE_TESTS_FLAG="-DADD_WINDOWS_ENCLAVE_TESTS=1"
 }
 
-& cmake.exe -G $BUILD_GENERATOR $LINUX_BIN_FLAG $ADD_WINDOWS_ENCLAVE_TESTS_FLAG $BUILD_ENCLAVES_FLAG ..
+# Create Build Type parameter
+if ($BUILD_TYPE -eq "Release") {
```

Comment:
It is not clear from this PR why CI related changes are needed.
Why does mbed test specifically need all these changes?

Reply:
The mbed test failed with the Win2016ReleaseCrossPlatform() CI run and Passed manually with all possible configurations. So, I trouble-shooted the CI PS script at length and uncovered that 
both the Release configurations never worked correctly. Debug modes worked with the 2 generators. The Release mode with one of the generators worked purely in Debug mode - host and enclaves were debug; with the other generator, it was kind of mixed mode; release enclaves and debug host; hence, the test mbed barfed where a debug test case was not created in Release mode. 

I cannot check in the mbed test without my fix for the Windows CI power-shell scripts.

Reply:
OK, here is the relevant snippet from the CI failure:
![image](https://user-images.githubusercontent.com/16311437/52095503-00b9f180-2578-11e9-92ee-0b699453b719.png)

Reply:
The debug.c test case is only relevant in Debug Mode, not Release mode.

Reply:
That CI fix seems like a very important fix that deserves its own PR.

Reply:
Yep, I thought so too but procrastinated. Filed a PR #1406 for the CI change just now.